core/pkg/evaluator/json.go

```diff
@@ -460,11 +467,26 @@ func configToFlags(log *logger.Logger, config string, newFlags *Flags) error {
 		return fmt.Errorf("transposing evaluators: %w", err)
 	}
 
-	err = json.Unmarshal([]byte(transposedConfig), &newFlags)
+	var configData ConfigWithMetadata
+	err = json.Unmarshal([]byte(transposedConfig), &configData)
 	if err != nil {
 		return fmt.Errorf("unmarshalling provided configurations: %w", err)
 	}
 
+	// Assign the flags from the unmarshalled config to the newFlags struct
+	newFlags.Flags = configData.Flags
+
+	// Assign metadata as a map to each flag's metadata
+	for key, flag := range newFlags.Flags {
+		if flag.Metadata == nil {
+			flag.Metadata = make(map[string]interface{})
+		}
+		for metaKey, metaValue := range configData.Metadata {
+			flag.Metadata[metaKey] = metaValue
+		}
+		newFlags.Flags[key] = flag
+	}
```

Comment:
`configData.Metadata` should have a lower merge priority than `flag.Metadata`.

Reply:
You can test the expected behavior in the [flagd playground](https://flagd.dev/playground/?scenario-name=Flag+metadata). This is using the JS implementation but the behavior should be the same. Thanks!